helmutils/helm_indent_finder_test.go

```diff
@@ -0,0 +1,244 @@
+			data := `
+roleRef:
+  kind: ClusterRole
+  name: gloo-graphqlapi-mutator-default
+  apiGroup: rbac.authorization.k8s.io
+---
+# Source: gloo-ee/charts/gloo/templates/1-gloo-deployment.yaml
+ 
+
+
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+`
+			badWindows := helmutils.FindHelmChartWhiteSpaces(data, opts)
+			Expect(len(badWindows)).To(Equal(1))
+		})
+		It("should detect a single white spaces in the yaml", func() {
+			opts := helmutils.HelmDetectOptions{DetectWhiteSpacesInEmptyLines: true}
+			data := `
+      
+      
+---
+
+---
+# Source: gloo-ee/charts/gloo/templates/2-gloo-service.yaml
+ 
+
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+`
+			badWindows := helmutils.FindHelmChartWhiteSpaces(data, opts)
+			Expect(len(badWindows)).To(Equal(2))
+		})
+	})
+
+	Describe("dealing with arrays in helmutils", func() {
+		It("should be ok with array in array", func() {
+			data := `
+  resources:
+  - federatedauthconfigs
+  - federatedauthconfigs/status
+  verbs:
+  - '*'
+- apiGroups:
+    - fed.ratelimit.solo.io
+  resources:
+    - federatedratelimitconfigs
+    - federatedratelimitconfigs/status
+  verbs:
+    - '*'
+`
+			badWindows := helmutils.FindHelmChartWhiteSpaces(data, opts)
+			Expect(len(badWindows)).To(Equal(0))
+		})
+		It("should find that the array is off by one", func() {
+			// the line fed.ratelimit.solo.io is off by one space
+			data := `
+  resources:
+  - federatedauthconfigs
+  - federatedauthconfigs/status
+  verbs:
+  - '*'
+- apiGroups:
+     - fed.ratelimit.solo.io
+  resources:
+    - federatedratelimitconfigs
+    - federatedratelimitconfigs/status
+  verbs:
+    - '*'
+`
+			badWindows := helmutils.FindHelmChartWhiteSpaces(data, opts)
+			Expect(len(badWindows)).To(Equal(1))
+		})
+		It("should find that the array is off by 4 spaces", func() {
+			// note the - "" is off by 4 spaces, this is acceptable
+			// this is whithin the array formatting specs. Same line, 2 spaces, or 4 spaces.
+			data := `
+resource:
+  release: gloo-ee-test
+    chart: prometheus-13.3.3
+    heritage: Helm
+  name: glooe-prometheus-server
+rules:
+  - apiGroups:
+      - ""
+    resources:
+      - nodes
+      - nodes/proxy
+      - nodes/metrics
+      - services
+`
+			badWindows := helmutils.FindHelmChartWhiteSpaces(data, opts)
+			Expect(len(badWindows)).To(Equal(0))
+		})
+		It("should accept arrays", func() {
+			data := `
+resources:
+  something:
+    there:
+      should:
+        be:
+          envFrom:
+            - configMapRef:
+                name: gloo-ee-test-observability-config
+            - secretRef:
+                name: gloo-ee-test-observability-secrets
+      securityContext:
+        runAsNonRoot: true
+        runAsUser: 10101
+`
+			badWindows := helmutils.FindHelmChartWhiteSpaces(data, opts)
+			Expect(len(badWindows)).To(Equal(0))
+		})
+	})
+
+	Describe("special breaks", func() {
+		DescribeTable("should accept special breaks", func(specialBreak string) {
```

Comment:
What happens where there's special breaks _and_ more data afterward? For example, adding more fields/items after the `envoy.json` one

Reply:
not sure what you mean.

You have the following, in which the `data` and `more data ...` are just a part of the special break. Could you show me what you mean?
```yaml
something: |-
    data..
     more data...
 something1: info
 ```

Reply:
I was thinking like
```
something: |-
  blah
     ijasd
  basu
someotherfield:
  - something
```

not sure if that's valid, but was wondering if we fully stopped checking the syntax after the special break, or if we continue checking different fields _outside_ that break. (if that makes sense)

Reply:
yeah this is covered.

`someotherfieldnotpartofabovebreak` will not cause an issue.
Updated the test, by adding this line.

Reply:
Oh just for clarification before resolving, if that `someotherfieldblahblah` field had an error, it would be caught right?

Reply:
if `someotherfieldblahblah` was off it would not produce an error. This would be because it would considered the same block as the special break.  It would be read like this in YAML

Reply:
`someotherfieldblahblah` would have to start on the same indent or lower as the one that started the special brake.